### DISCUSSION TITLE
How can i subscribe to multiple rows?

### DISCUSSION THREAD

**Discussion by farzd** [supabase/supabase#7735] [Questions]
I have multiple id's i want to subscribe to. Is there for me to use a OR operand when subscribing ? or do i need to do separate subscriptions for each? 

```
//can i do something like this?
 supabasefrom(`chats:id=eq.${ids[0]},id=eq{ids[1]}`)on('UPDATE', (payload) => {
    console.log({ payload })
})
```
Use case is tinder like chat page
![image](https://user-images.githubusercontent.com/1423413/179400228-4d0ba715-81c1-4c67-ba94-5f74aa2df80d.png)

**Comment by GaryAustin1** [comment]
Currently you need separate subscription for each.  There is some discussion on an in filter which would solve this, but I’ve seen nothing on a time frame.

**Reply by farzd** [reply]
Ok Thanks man, been following your threads about subscriptions - appreciate the work you've put in with your findings.
I may have to subscribe for each chat thread but not sure how much db performance will degrade as there could well be crazy amount of subscriptions per user.

**Reply by GaryAustin1** [reply]
I've gone to a split method for my planned chat area in my app (low volume chats, not a key part of app).  When the user is doing other things and not in a room, I poll the user's chat room rows (they contain last updated time) every 5 minutes and compare against last update.  If the user enters a chat room then I turn on realtime for just that chatroom.
I think this is not much different overhead than have the multiple realtime connections going with data every 30 seconds for heartbeat, keeping them refreshed when tabs change or device goes in power save mode (which involves doing the same fetches as my polling).

**Reply by farzd** [reply]
Thats a great idea, my realtime does connect when they are in the chat room but its the listing [tinder screenshot] which would required a lot of concurrent subscriptions. Right now that list is refreshed when the app is brought to foreground or when you tab to the listing page. But if you stay on the page then i guess polling is a really good shout.

i'm trying to implement your flow chart for reconnecting any lost subscriptions but logs on mobile apps are not very persistent when its in the background and hard for me to see if the connection is lost, sometimes the subscription .status [subscribe method] doesn't report.

**Reply by GaryAustin1** [reply]
I'm trying to do some updates on that flow chart and code in a repository over the next couple of weeks or so.  In particular the visibility handler may be problematic now as SB added a visibility handler to refresh the token a month or so ago and I think it matters which one runs first, yet there is no way to control that.   Basically though just assume if your tab is not in focus, realtime could lose connection and not let you know.  I'm trying to optimize standard desktop browsers a bit as they have 5 minutes before they start losing connection.

**Reply by farzd** [reply]
Yeah i'm going with that assumption of losing connection if its out of focus. Although so far its maintained connection when app is in BG for over 15+ minutes, but that is device dependant.
i've set my JWT's to be 604800, so it lasts a week, so not sure my issue is related to the refreshtoken.

**Reply by timothygorer** [reply]
Hi guys, I was wondering how I could write a for loop where each iteration of the loop creates a new subscription. I have a flatlist in my react native app where each card in the flatlist has a different value for the 'name' column in a supabase table called Games. I would like to listen to changes for each of these cards. I have tried doing this for loop, but the subscribe() is not working. It only works if I individually call .subscribe() without a for loop. @farzd @GaryAustin1 Thanks.
